client/components/SingleModel.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import { connect } from 'react-redux';
+
+class SingleModel extends React.Component{
+    constructor(){
+        super()
+        this.state = {
+            model:'',
+            description:'',
+            price:0,
+            image:''
+        }
+    }
+    componentDidMount(){
+        const model = this.props.products.find(e => e.id === this.props.match.params.id*1)
+        this.setState({
+            model: model.model,
+            description: model.description,
+            price: model.price,
+            image: model.image
+        })
```

Comment:
Since we're just grabbing properties with the same name as their equivalent on state off of the model object here it might be worth thinking about if there's a cleaner way to write this